File: modules/book/src/main/java/me/ag2s/base/ThrowableUtils.java
package me.ag2s.base;

import androidx.annotation.NonNull;

import java.io.IOException;

public class ThrowableUtils {


    public static @NonNull
    IOException rethrowAsIOException(Throwable throwable) throws IOException {
        throw new IOException(throwable.getMessage(), throwable);
    }
}

File: app/src/main/java/io/legado/app/lib/epublib/base/ThrowableUtils.kt
package io.legado.app.lib.epublib.base

import java.io.IOException

object ThrowableUtils {
    @Throws(IOException::class)
    fun rethrowAsIOException(throwable: Throwable): IOException {
        val newException = IOException(throwable.message, throwable)
        throw newException
    }
}
